tcslackbuildnotifier-core/src/main/java/slacknotifications/SlackNotificationImpl.java

```diff
@@ -314,31 +366,88 @@ private void postViaWebHook() throws IOException {
+                    numUserCommits++;
+                }
+            }
+            numUserCommitsLeft = numUserCommits;
+
+            if (numUserCommits > maxCommitsToDisplay) {
+                truncatedUser = true;
+            }
+
+            if (totalCommits > maxCommitsToDisplay && totalCommits > numUserCommits) {
+                //commitsToDisplay = commitsToDisplay.subList(0, maxCommitsToDisplay > commitsToDisplay.size() ? commitsToDisplay.size() : 5);
+                truncatedOther = true;
             }
 
             for (Commit commit : commitsToDisplay) {
                 String revision = commit.getRevision();
                 revision = revision == null ? "" : revision;
-                sbCommits.append(String.format("%s :: %s :: %s\n", revision.substring(0, Math.min(revision.length(), 10)), commit.getUserName(), commit.getDescription()));
+                if(commit.getSlackUserId().equals(slackUser.substring(1))){
+                    sbUserCommits.append(String.format("%s :: %s :: %s\n", revision.substring(0, Math.min(revision.length(), 10)), commit.getUserName(), commit.getDescription()));
+                    numUserCommitsLeft--;
+                    numCommitsToAdd--;
+                }
+                else{
+                    if(numUserCommitsLeft < numCommitsToAdd){
+                        sbCommits.append(String.format("%s :: %s :: %s\n", revision.substring(0, Math.min(revision.length(), 10)), commit.getUserName(), commit.getDescription()));
+                        numCommitsToAdd--;
+                    }
+                }
+
+                if(numCommitsToAdd == 0){
+                    break;
+                }
+            }
+
+            if (truncatedOther && !truncatedUser) {
+                sbCommits.append(String.format("(+ %d more)\n", totalCommits - maxCommitsToDisplay));
+            }
+
+            if (truncatedUser) {
+                sbUserCommits.append(String.format("(+ %d more)\n", numUserCommits - maxCommitsToDisplay));
             }
 
-            if (truncated) {
-                sbCommits.append(String.format("(+ %d more)\n", totalCommits - 5));
+            if (sbUserCommits.length() > 0) {
+                attachment.addField("Your Commits", sbUserCommits.toString(), false);
             }
 
-            if (!commitsToDisplay.isEmpty()) {
+            if (sbCommits.length() > 0) {
                 attachment.addField("Commits", sbCommits.toString(), false);
             }
+            else if(totalCommits > numUserCommits){
+                String otherCommits;
+                if(totalCommits - numUserCommits > 1){
+                    otherCommits = totalCommits - numUserCommits + " more commits were made by others";
+                }
+                else{
+                    otherCommits = "1 more commit was made by others";
+                }
+                attachment.addField(otherCommits, null, false);
+
+            }
+
+
```

Comment:
`getAttachements()` is hard to read and understand. There are many mutable variables, conditionals, and the method is more than 100 lines long (not always bad, but ideally a method should fit on screen and be short). This method needs to be restructured.
Let's discuss ways to restructure this.